enhancements/machine-api/machine-api-usage-telemetry.md

```diff
@@ -0,0 +1,244 @@
+and insight into infrastructure stability.
+
+## Motivation
+
+Build an understanding of how frequently Machine API components are being used,
+and provide a window into the general health of clusters based on failure and
+remediation rates.
+
+### Goals
+
+* Inform about Machine API component(Machine API Operator, Cluster Autoscaler,
+and Machine Health Check) usage across the OpenShift fleet.
+* Expose Machine Health Check success and failure rates to provide data on
+remediation effectiveness and general cluster health trends.
+* Create alerts that detail specific failure modes and remediations that are
+occurring within OpenShift to capture events related to Machine API function.
+
+### Non-Goals
+
+* Provide any information that would expose a user's identify or confidential
+information.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an OpenShift developer I would like to improve my understanding of how users
+are consuming the Machine API and its related components. By creating time
+series telemetry about the number and type of deployed component resources
+(for example MachineSets) I will learn about usage patterns and trends. Learning
+more about how frequently these components are deployed and utilized will allow me to
+identify areas where users might be having difficulties, make better decisions
+about the future planning of work, and demonstrate to a wider audience the popularity
+level of various features.
+
+#### Story 2
+
+As an OpenShift developer I would like to gain a deeper understanding of the
+general health of clusters by examining the behavior of the Machine Health
+Check. By analyzing the frequency of success and failures rates as well as
+the operational modes of the Machine Health Check I will be able to build
+a model of infrastructure health and potential predictive activity. This
+information will allow me to gain insight into correlations between
+unhealthy cluster remediations and infrastructure providers, give a window
+into predictive possibilities for user outreach, and demonstrate the value of
+automated remediation.
+
+#### Story 3
+
+As an OpenShift developer I would like to correlate Machine API resource usage
+with alerting events to provide presciptive advice about failures conditions
+and platform instability. By examining data about Machine API resource management,
+failures, and remediations, and corelating that data with alerts and platform
+information I will be able to better advise users about errors and methods for
+avoiding them. When combined with a wider fleet view this information will help
+me to create advisories about provider instabilities and potential outtages.
+
+### Implementation Details/Notes/Constraints
+
+#### Alerts, Metrics, and Telemetry
+
+To fully accomplish the goals proposed in this enhancement several metrics
+will need to be exposed through telemetry. These metrics are:
+
+* Machine resource count
```

Comment:
This IMO is redundant to nodes.

Reply:
For UPI environments automated machine management via machine API is a choice. I think the relation between number of machines and node could give us meaningful insight about the machine API usage in such environments.
I think this could also be valuable in a cluster created via IPI to understand if user cares at all about failed machines. e.g ~100 machines but only ~10nodes. And so identify environments where MHC would come in handy.

Reply:
Yeah, it is not redundant to nodes.  Machines and nodes are not the same.